src/transaction/components/TransactionButton.tsx

```diff
@@ -7,12 +7,27 @@ export function TransactionButton({
   className,
   text: buttonText = 'Transact',
 }: TransactionButtonReact) {
-  const { address, contracts, isLoading, onSubmit, transactionId } =
-    useTransactionContext();
+  const {
+    address,
+    contracts,
+    errorMessage,
+    isLoading,
+    onSubmit,
+    status,
+    transactionHash,
+    transactionId,
+  } = useTransactionContext();
 
-  // TODO: should disable if transactionId exists ?
   const isDisabled = isLoading || !contracts || !address || transactionId;
 
+  // there is a delay between when txn is confirmed and
+  // call status returns isLoading true so the third
+  // condition here keeps spinner visible during that time
```

Comment:
Loooove this comment.

Reply:
Btw when I see too many conditions I always think about, how to create an utility `get` or `hook` from it.

That's just because are easier to unit test and mantain.

Something as follow up.